mesheryctl/pkg/utils/scripts.go

```diff
@@ -6,118 +6,250 @@ import (
 	"os/exec"
 )
 
-// TODO@Aisuko this function is so hard to read and maintain, please refactor it
-func GenerateConfigGKE(configPath, SAName, namespc string) error {
-	script := fmt.Sprintf(`
-	set -e
-	set -o pipefail
-
-	KUBECFG_FILE_NAME="%s"
-	SERVICE_ACCOUNT_NAME=%s
-	NAMESPACE="%s"
-	TARGET_FOLDER=$(dirname ${KUBECFG_FILE_NAME})
-
-	command_exists() {
-		command -v $1 >/dev/null 2>&1
+// GKEConfig holds configuration for GKE script generation
+
+type GKEConfig struct {
+	ConfigPath string
+	SAName     string
+	Namespace  string
+}
+
+// GenerateConfigGKE generates kubernetes config for GKE
+func GenerateConfigGKE(configPath, SAName, namespace string) error {
+	cfg := &GKEConfig{
+		ConfigPath: configPath,
+		SAName:     SAName,
+		Namespace:  namespace,
 	}
-	check_prerequisites() {
-		if ! command_exists jq ; then
-			echo -e "\\nMissing required utility: 'jq'. Please install 'jq' and try again."
-			exit;
-		fi
-		if ! command_exists base64 ; then
-			echo -e "\\nMissing required utility: 'base64'. Please install 'base64' and try again."
-			exit;
-		fi
-		if ! command_exists awk ; then
-			echo -e "\\nMissing required utility: 'awk'. Please install 'awk' and try again."
-			exit;
-		fi
-		if ! command_exists tail ; then
-			echo -e "\\nMissing required utility: 'tail'. Please install 'tail' and try again."
-			exit;
-		fi
+
+	if err := cfg.validate(); err != nil {
+		return fmt.Errorf("validation failed: %w", err)
```

Comment:
wondering if it should return mesheryctl err

Reply:
I used `fmt.Errorf` to improve output formatting. Let me know if your opinion , I'm happy to adjust as needed.

Reply:
<img width="461" alt="Screenshot 2024-11-15 at 12 50 11 PM" src="https://github.com/user-attachments/assets/bb6f77ff-c4b8-4f07-98e3-00a9225d8f30">
i guess using this should be better ??